source/Plugins.cpp

```diff
@@ -0,0 +1,135 @@
+/* Plugins.cpp
+Copyright (c) 2022 by Sam Gleske (samrocketman on GitHub)
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "Plugins.h"
+
+#include "DataFile.h"
+#include "DataNode.h"
+#include "DataWriter.h"
+#include "Files.h"
+
+#include <algorithm>
+#include <map>
+
+using namespace std;
+
+namespace {
+	// The first item in pair is plugin state on load, second item is user changing preference.
+	map<string, pair<bool, bool>> settings;
+
+	void LoadSettings(const string &path)
+	{
+		DataFile prefs(path);
+		for(const DataNode &node : prefs)
+		{
+			const string &key = node.Token(0);
+			if(key == "plugins")
+				for(const DataNode &child : node)
+					if(child.Size() == 2)
+						settings[child.Token(0)].first = child.Value(1);
```

Comment:
Any reason not to set both bools here?
```suggestion
						settings[child.Token(0)] = make_pair(child.Value(1), child.Value(1));
```
I'm not sure if there is a performance penalty to calling `child.Value(1)` twice, the compiler might just cache it (unless it doesn't know that the return value won't change, I suppose that depends on if `child.Value()` is `const`).
```suggestion
					{
						bool state = child.Value(1);
						settings[child.Token(0)] = make_pair(state, state);
					}
```

Reply:
Resolved by using tehhowch struct with support for setting the initial state via constructor.